### ISSUE TITLE
[Q] How does The network ensemble deal with variable number of frames in RNNBeatProcessor?

### ISSUE THREAD

**Issue by siryog90** [CPJKU/madmom#307]
@superbock 
Sorry to bother, but one last detail I need to understand is how your saved neural network model deals with the fact that the number of Spectrogram frames depends on the length of the song in question. Even the training data set from MIREX 2006 varies in the number of milliseconds so the number of Spectrogram frames is not exactly the same for all of them. I just zero them to make the sequences uniform length, but `RNNBeatProcessor` class doesn't seem to zero pad the parameter signal.

Also, with `fps=100` I get close to 3000 frames for a 30 second audio excerpt. Some of the label files contain 10 to 20 beat time measurements. In other words, 0.05% of the data is labeled as 'beat'. Seems like a minuscule number for a neural net to learn. How did you deal with this?

**Comment by superbock** [comment]
The sequence length doesn't matter for RNNs -- nor does it for any other network. The RNN simply processes the whole input sequence frame by frame.

**Comment by siryog90** [comment]
@superbock 
Makes sense. Already fixed that issue.

What about the second point? If for example 'beat' is labeled at time 2012 milliseconds, did you label a whole second's worth of frames as 'beat'?

i.e. if 'beat' is at 2012 milliseconds then, at 100 fps (a frame for every 10 milliseconds; 1 ~ 100 frames), `frames[2012:2112] = 1`

**Comment by superbock** [comment]
The second point wasn't there when I replied, please use the edit function only for minor edits (e.g. spelling).

The beat targets are only set to '1' for the frame the beat occurs/is annotated. Yes, this is a small fraction of all frames, but it is much closer to 5% -- not 0.05%. If we assume a tempo of 120 bpm, there are two beats per second, i.e. 2 frames out of 100 are annotated as beats. Cross entropy error deals exactly with this sort of unbalanced targets.

**Comment by superbock** [comment]
Closing after 18 days of inactivity.
